### PR TITLE
Add callback doc

### PR DESCRIPTION
**Pull request by yzhang-23**
**Is this pull request associated with an issue(s)?**
This PR is to complete the work in [PR 128](https://github.com/NWChemEx/ParallelZone/pull/128).

**Description**
Design documentation on finalize callbacks in the RuntimeView class is added. A simple example on how to set the callbacks is also presented.

**TODOs**
[X] To test callbacks in downstream repos.

### CODE REVIEW

File: docs/source/developer/design/assets/runtime_view.png
Comment (ryanmrichard) [inline docs/source/developer/design/assets/runtime_view.png:1]: This image doesn't seem to contain the Excalidraw scence. Please remake with the "MPI_Init()" box just saying "finalize callbacks" and the caption saying "ties lifetime of other parallel runtimes to the RuntimeView"

**Review by ryanmrichard — CHANGES_REQUESTED** [review]

File: docs/source/developer/design/assets/runtime_view.png
Comment (yzhang-23) [reply docs/source/developer/design/assets/runtime_view.png:1]: Sorry, what do you mean for not "contain the Excalidraw scence" (what is missing?)?

Comment (ryanmrichard) [reply docs/source/developer/design/assets/runtime_view.png:1]: Try opening `runtime_view.png` in Exacalidraw. I get the error:
> This image does not seem to contain any scene data. Have you enabled scene embedding during export?

**Review by ryanmrichard — APPROVED** [review]

**Review by jwaldrop107 — APPROVED** [review]

**Comment by ryanmrichard** [comment]
🚀 [[bumpr]](https://github.com/haya14busa/action-bumpr) [Bumped!](https://github.com/NWChemEx/ParallelZone/actions/runs/7659786673) **New version**:[v0.1.22](https://github.com/NWChemEx/ParallelZone/releases/tag/v0.1.22) **Changes**:[v0.1.21...v0.1.22](https://github.com/NWChemEx/ParallelZone/compare/v0.1.21...v0.1.22)

**Pull request merged**